pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -470,6 +470,9 @@ func realScheduleMonitor(c *Cluster, mon *monConfig) (SchedulingResult, error) {
 		d.Spec.Template.Spec.Volumes = append(d.Spec.Template.Spec.Volumes,
 			controller.DaemonVolumesDataPVC(mon.ResourceName))
 		controller.AddVolumeMountSubPath(&d.Spec.Template.Spec, "ceph-daemon-data")
+
+		// Set the PVC label name so that it can be deleted on cleanup
+		d.Labels["pvc_name"] = pvcName
```

Comment:
It would be preferred if the pvc name was available above so the label could be set immediately inside the call to `makeDeployment()`, then `getLabels()` could set the pvc_name label directly. This way, it will be overridden after the fact. But changing the order of creating the pvc first would be a much bigger change, so let's keep it with this simple change.

Reply:
You may also want to set it on the pod spec labels, even if not strictly necessary.

Reply:
Actually I wonder if this will cause the PVCs to be deleted every time a mon is created, which would be bad if the PVC were deleted while the main mon deployment is using it. `removeCanaryDeployments()` is always called after a new mon is started, which would mark the pvc as deleted. We only want to delete the PVC if the new mon failed to be added to quorum.